driver-core/src/main/com/mongodb/operation/ListCollectionsOperation.java

```diff
@@ -186,7 +235,24 @@ private MongoNamespace getNamespace() {
     }
 
     private BsonDocument getCommand() {
-        return new BsonDocument("listCollections", new BsonInt32(1)).append("cursor", new BsonDocument());
+        BsonDocument cursorDocument = batchSize == null ? new BsonDocument() : new BsonDocument("batchSize", new BsonInt32(batchSize));
+        BsonDocument command = new BsonDocument("listCollections", new BsonInt32(1)).append("cursor", cursorDocument);
+        if (filter != null) {
+            command.append("filter", filter);
+        }
+        return command;
+    }
+
+    private BsonDocument getLegacyFilter() {
```

Comment:
Can you explain what we're trying to accomplish here?  Does the query filter work differently for the command and the legacy query?